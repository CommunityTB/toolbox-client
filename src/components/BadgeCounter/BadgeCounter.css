.tools_count {
  display: block;
  position: absolute;
  top: 34px;
  right: 8px;
  /* line-height: 14px; */
  height: 18px;
  padding: 0 5px;
  font-family: Arial, sans-serif;
  font-size: 14px;
  color: white;
  text-shadow: 0 1px rgba(0, 0, 0, 0.25);
  border: 1px solid;
  border-radius: 10px;
  box-shadow: inset 0 1px rgba(255, 255, 255, 0.3), 0 1px 1px rgba(0, 0, 0, 0.08);
  background: #FBB456;
  border-color: #FBB456;
}