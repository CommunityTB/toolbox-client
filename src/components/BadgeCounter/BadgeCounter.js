import React from 'react';
import './BadgeCounter.css';

export default function BadgeCounter(props){
  const badge = props.count ? <div className="tools_count">{props.count}</div> : null;
  return (
    <div>
      {badge}
    </div>
  );
}