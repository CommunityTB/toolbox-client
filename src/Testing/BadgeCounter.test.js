import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import BadgeCounter from '../components/BadgeCounter/BadgeCounter'

describe(`BadgeCounter component`, () => {
  const props = {
    count: 1
  }
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BadgeCounter />, div);
    ReactDOM.unmountComponentAtNode(div);
  });


  it('renders a BadgeCounter by default', () => {
    const wrapper = shallow(<BadgeCounter />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the badge counter from props', () => {
    const wrapper = shallow(<BadgeCounter {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})