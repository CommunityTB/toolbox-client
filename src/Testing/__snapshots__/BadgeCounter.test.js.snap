// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeCounter component renders a BadgeCounter by default 1`] = `<div />`;

exports[`BadgeCounter component renders the badge counter from props 1`] = `
<div>
  <div
    className="tools_count"
  >
    1
  </div>
</div>
`;
